### PR TITLE
fix: on createDocument, return copy in _docmap

### PR DESCRIPTION
**Pull request by msterle**
Ensure that even if a document already existed, we always refer to the same one.

Had added cleanup code, but removed because the dispatcher assumes only one document per docId, so cleaning up one copy remove all copies from the dispatcher's list.

### CODE REVIEW

**Comment by oed** [comment]
Lmk if you need any pointers to fix the tests.

File: packages/ceramic-cli/package.json [hunk 1]
```diff
@@ -39,12 +39,12 @@
...
   "devDependencies": {
-    "@babel/core": "^7.8.4",
+    "@babel/core": "^7.9.0",
     "@babel/preset-env": "^7.8.4",
-    "@babel/preset-typescript": "^7.8.3",
+    "@babel/preset-typescript": "^7.9.0",
     "@types/events": "^3.0.0",
     "@types/express": "^4.17.2",
-    "@types/node": "^13.7.4",
+    "@types/node": "^13.9.5",
```
Comment (oed) [inline packages/ceramic-cli/package.json:47]: ty!

File: packages/ceramic-cli/src/__tests__/ceramic-daemon.test.ts [hunk 2]
```diff
@@ -64,17 +64,10 @@ describe('Ceramic interop: core <> http-client', () => {
...
-    expect(doc1.state.log.length).toEqual(2)
-    expect(doc1.state.anchored).toBeGreaterThan(0)
-    const doc2 = await client.createDocument({ test: 123 }, DOCTYPE_TILE)
-    await anchorUpdate(doc2)
-    expect(doc2.state.log.length).toEqual(2)
-    expect(doc2.state.anchored).toBeGreaterThan(0)
+    const doc = await core.createDocument({ test: 123 }, DOCTYPE_TILE)
+    await anchorUpdate(doc)
+    expect(doc.state.log.length).toEqual(2)
+    expect(doc.state.anchored).toBeGreaterThan(0)
```
Comment (oed) [inline packages/ceramic-cli/src/__tests__/ceramic-daemon.test.ts:70]: Actually can you just add a test for the `client` that creates a different document?

File: packages/ceramic-http-client/package.json [hunk 3]
```diff
@@ -27,19 +27,19 @@
     "clean": "rm -rf ./lib"
   },
   "dependencies": {
+    "@types/node-fetch": "^2.5.5",
```
Comment (oed) [inline packages/ceramic-http-client/package.json:30]: This should be in dev-deps.

File: packages/3id-did-resolver/package.json [hunk 4]
```diff
@@ -45,5 +44,8 @@
     "transformIgnorePatterns": [
       "/node_modules(?!/did-jwt)/"
     ]
+  },
+  "dependencies": {
+    "did-resolver": "^1.1.0"
```
Comment (oed) [inline packages/3id-did-resolver/package.json:49]: Why was this moved to regular deps?

File: packages/ceramic-http-client/package.json (hunk 3 above)
Comment (msterle) [reply packages/ceramic-http-client/package.json:30]: We use `node-fetch` in production code for this package, so the types are needed as well

File: packages/3id-did-resolver/package.json (hunk 4 above)
Comment (msterle) [reply packages/3id-did-resolver/package.json:49]: https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/3id-did-resolver/src/index.ts#L1

File: packages/ceramic-http-client/package.json (hunk 3 above)
Comment (oed) [reply packages/ceramic-http-client/package.json:30]: hm, not sure I follow. The types should only be needed when compiling?

File: packages/3id-did-resolver/package.json (hunk 4 above)
Comment (oed) [reply packages/3id-did-resolver/package.json:49]: We only use the types so we shouldn't need to have it as a dep. What command are you using the `--production` flag on? And when do we need to do that? Usually we build before we publish to npm.

Comment (oed) [reply packages/3id-did-resolver/package.json:49]: Still confused by this.

Comment (msterle) [reply packages/3id-did-resolver/package.json:49]: @oed  Without this, when the package is imported into `3box-js` and the example server is run, I get an error in the browser
```
Uncaught (in promise) TypeError: did_resolver_1.Resolver is not a constructor
    at new TileHandler (tileHandler.js:29)
    at new Ceramic (ceramic.js:39)
    at Function.eval (ceramic.js:59)
    at Generator.next (<anonymous>)
    at eval (ceramic.js:8)
    at new Promise (<anonymous>)
    at __awaiter (ceramic.js:4)
    at Function.create (ceramic.js:57)
    at Box._callee$ (3box.js:155)
    at tryCatch (runtime.js:45)
```

This is possibly because of the different versions of `did-resolver` used by 3box-js and ceramic. It might be only because I am volume mounting the package instead of using a published version, but since this was moved into dev dependencies since the last published version, it looked like a mistake. Can you please publish the latest version so we can test it?

Comment (msterle) [reply packages/3id-did-resolver/package.json:49]: Nevermind, this error was being caused by the combination of lerna hoisting and docker volume mounting - just not going to hoist in local development.

Comment (oed) [reply packages/3id-did-resolver/package.json:49]: ok, I guess we can move it to here for now then. Let's merge this then make a new release!
